CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+cmake_minimum_required(VERSION 3.5)
+
+# ouptut paths
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_LIBDIR}")
+
+# Build shared libs
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
+if(MSVC)
+    set(CMAKE_DEBUG_POSTFIX "d")
+endif()
+
+option(BUILD_SHARED_LIBS "Build libraries as shared as opposed to static" ON)
+
+list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+# Enable C++14
+set(CMAKE_CXX_EXTENSIONS OFF)
+set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+# include macros for warnings
+include(AddWarningsConfigurationToTargets)
+include(CMakePackageConfigHelpers)
+
+project(levi
+        LANGUAGES CXX
+        VERSION 0.0.100)
+
+include(GNUInstallDirs)
+
+option(ENABLE_RPATH "Enable RPATH for this library" ON)
+mark_as_advanced(ENABLE_RPATH)
+include(AddInstallRPATHSupport)
+add_install_rpath_support(BIN_DIRS "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_BINDIR}"
+                          LIB_DIRS "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+                          DEPENDS ENABLE_RPATH
+                          USE_LINK_PATH)
+
+# Encourage user to specify a build type (e.g. Release, Debug, etc.), otherwise set it to Release.
+if(NOT CMAKE_CONFIGURATION_TYPES)
+  if(NOT CMAKE_BUILD_TYPE)
+      message(STATUS "Setting build type to 'Release' as none was specified.")
+      set_property(CACHE CMAKE_BUILD_TYPE PROPERTY VALUE "Release")
+  endif()
+endif()
+
+find_package(Eigen3 REQUIRED)
+
+add_library(levi INTERFACE)
+target_include_directories(levi INTERFACE ${EIGEN3_INCLUDE_DIR})
+target_include_directories(levi INTERFACE $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>)
+target_include_directories(levi INTERFACE $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
+
+
+# install
+install(TARGETS levi
+        EXPORT levi
+        COMPONENT runtime
+        RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}" COMPONENT bin
+        LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT shlib
+        ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}" COMPONENT lib
+        PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+
+install(EXPORT levi
+        DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/levi)
```

Comment:
I strongly suggest to add a NAMESPACE with the same name of the CMake package (see https://github.com/robotology/how-to-export-cpp-library/blob/782e9b1b13867686463b2a8a5b2b9d67aa176099/cmake/InstallBasicPackageFiles.cmake#L673) so that your target will be usable as `levi::levi`.

Reply:
Is it enough to add ``NAMESPACE levi::`` in the ``install`` command?

Reply:
Yes. 

I also suggest to add a `levi::levi` ALIAS to the target of the library, see https://github.com/robotology/how-to-export-cpp-library/blob/e23e399fce102ba8f87cd5269797513000b99794/src/LibTemplateCMake/CMakeLists.txt#L62 . This simplifies the process of consuming the library with [`FetchContent`+`add_subdirectory`](https://cmake.org/cmake/help/v3.11/module/FetchContent.html), that is quite fashionable these in CMake world.